js/parser.js

```diff
@@ -54,22 +54,58 @@ var ERMrest = (function(module) {
             baseUri: uri.slice(0,svc_idx+_service_name_len)
         };
         var path = context.path = uri.slice(svc_idx+_service_name_len); // string after service
+        if (uri.indexOf("@sort(") !== -1) {
+            context.uri = uri.split("@sort(")[0]; // remove modifiers from uri
+        } else if (uri.indexOf("@before(") !== -1) {
+            context.uri = uri.split("@before(")[0]; // remove modifiers from uri
+        } else if (uri.indexOf("@after(") !== -1) {
+            context.uri = uri.split("@after(")[0]; // remove modifiers from uri
+        } else if (uri.indexOf("?limit=") !== -1) {
+            context.uri = uri.split("?limit=(")[0]; // remove modifiers from uri
+        }
+
+        var modifierPath = uri.split(context.uri)[1];
 
         // Parse out @sort(...) parameter and assign to context
         // Expected format:
         //  ".../catalog/catalog_id/entity/[schema_name:]table_name[/{attribute::op::value}{&attribute::op::value}*][@sort(column[::desc::])]"
-        if (uri.indexOf("@sort(") !== -1) {
+        if (modifierPath) {
+            if (modifierPath.indexOf("@sort(") !== -1) {
+
+                var sorts = modifierPath.match(/@sort\(([^\)]*)\)/)[1].split(",");
+                path = path.split("@sort(")[0];  // anything before @sort(..)
+
+                context.sort = [];
+                for (var s = 0; s < sorts.length; s++) {
+                    var sort = sorts[s];
+                    var column = (sort.endsWith("::desc::") ?
+                        decodeURIComponent(sort.match(/(.*)::desc::/)[1]) : sort);
+                    context.sort.push({"column": column, "descending": sort.endsWith("::desc::")});
+                }
+            }
 
-            var sorts = path.match(/@sort\((.*)\)/)[1].split(",");
-            path = path.split("@sort(")[0];  // anything before @sort(..)
-            context.uri = uri.split("@sort(")[0]; // remove @sort from uri
+            // sort must specified to use @before and @after
+            if (modifierPath.indexOf("@before(") !== -1) {
+                if (context.sort) {
+                    context.paging = {};
+                    context.paging.before = true;
+                    context.paging.row = {};
+                    var row = modifierPath.match(/@before\(([^\)]*)\)/)[1].split(",");
+                    for (var i = 0; i < context.sort.length; i++) {
+                        context.paging.row[context.sort[i].column] = decodeURIComponent(row[i]);
```

Comment:
You also need to handle the special value `::null::` here and below for each paging modifier component, before decoding the strings.  This special value indicates an actual `null` for the page key component, while an empty component would be the empty string `""`.